pkgs/development/tools/build-managers/meson/default.nix

```diff
@@ -101,6 +105,28 @@ python3.pkgs.buildPythonApplication rec {
     installShellCompletion --bash data/shell-completions/bash/meson
   '';
 
+  # while building documentation meson may want to run binaries for host
+  # which needs an emulator
+  # example of an error which this fixes
+  # [Errno 8] Exec format error: './gdk3-scan'
+  passthru.crossEmulatorWrapper =
+    let
+      crossFile = writeText "cross-file.conf" ''
+        [binaries]
+        exe_wrapper = ${lib.escapeShellArg (stdenv.targetPlatform.emulator buildPackages)}
+      '';
+      crossHook = writeText "crossHook" ''
+        add_meson_exe_wrapper_cross_flag() {
+          mesonFlagsArray+=(--cross-file=${crossFile})
+        }
+        preConfigureHooks+=(add_meson_exe_wrapper_cross_flag)
+      '';
+    in
+
+    if (stdenv.buildPlatform != stdenv.hostPlatform) then
+      makeSetupHook { name = "mesonCrossEmulatorWrapper"; } crossHook
+    else "";
```

Comment:
Maybe make this `throw` so that people do not accidentally introduce empty string to inputs, which could be an issue if we ever decide to be more strict about inputs (cannot find the relevant PR now).

Reply:
done and added a example to the docs